pkg/scheduler/dependency_graph.go

```diff
@@ -338,29 +368,44 @@ func (rl *ReplicaList) Swap(i, j int) {
 	rl.Items[i], rl.Items[j] = rl.Items[j], rl.Items[i]
 }
 
-func (sched *Scheduler) allocateReplicas(flow *client.Flow, options interfaces.DependencyGraphOptions) ([]string, error) {
-	label := labels.Set{"flow": flow.Name}
+func (sched *Scheduler) allocateReplicas(flow *client.Flow, options interfaces.DependencyGraphOptions) ([]client.Replica, []client.Replica, error) {
+	name := flow.Name
+	if options.FlowInstanceName != "" {
+		name = options.FlowInstanceName
+	}
+	label := labels.Set{"flow": name}
 	var tmp ReplicaList
```

Comment:
Yeah, i know about this types distinction (this is why we need casting to ReplicaList) but i'm proposing swap of names, as we there have a temporary data got from Replicas().List() which then after conversion is used in rest, bigger scope of func, after the cast.
The best would be to use `clientExistingReplicas` and `existingReplicas` and not using the name `tmp` at all - it's worthless there.